### DISCUSSION TITLE
Permission Denied for "filesystem" Notification Provider Default Instructions

### DISCUSSION THREAD

**Discussion by mc962** [authelia/authelia#7379] [Troubleshooting] [answered]
I wanted to experiment with Authelia and try things out using the `filesystem` Notification Provider. When following the default instructions pointing to a file at `/config/notification.txt`, I seem to be getting permission issues with the user (`authelia`) accessing the file when starting the system with `systemd`.
`Jun 01 22:05:34 authelia authelia[2384]: time="2024-06-01T22:05:34Z" level=error msg="Error occurred running a startup check" error="stat /config/notification.txt: permission denied" >`
I also tried changing the path to `/home/authelia/notification.txt` as I figured that Authelia should have permissions for files in that directory, but got a similar issue. The authelia user is part of the `sudo` group as well.

I tried both with the file created, and without it being created (in case it is expected that Authelia creates and manages the file itself). I also tried changing ownership and permissions to ensure that the authelia user owned the file and could read/write the file.

I followed the "bare-metal" installation instructions, and installed Authelia from the repository on a Debian LXC container on my Proxmox machine. Everything else seems to have started correctly according to the logs except for the Notification Provider.

I'm not sure if this is a bug, incorrect documentation, or something missing that I need to do to get this working properly, but I haven't seen others having quite the same issue so it feels like i'm missing some step somewhere.

**Comment by glime-ai** [comment, marked as answer]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It seems like you are encountering permission issues with the `filesystem` Notification Provider in Authelia. Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Check File Permissions and Ownership:**
   Ensure that the `notification.txt` file has the correct permissions and ownership. The file should be readable and writable by the `authelia` user.

   ```sh
   sudo chown authelia:authelia /config/notification.txt
   sudo chmod 600 /config/notification.txt
   ```

2. **Verify Directory Permissions:**
   Ensure that the directory containing the file also has the correct permissions and ownership.

   ```sh
   sudo chown -R authelia:authelia /config
   sudo chmod -R 700 /config
   ```

3. **Systemd Service File:**
   Ensure that the systemd service file for Authelia is correctly configured to run as the `authelia` user. The service file should include the `User` and `Group` directives.

   Example of a systemd service file (`/etc/systemd/system/authelia.service`):

   ```ini
   [Unit]
   Description=Authelia
   After=network.target

   [Service]
   User=authelia
   Group=authelia
   ExecStart=/usr/local/bin/authelia --config /etc/authelia/configuration.yml
   Restart=always

   [Install]
   WantedBy=multi-user.target
   ```

4. **SELinux/AppArmor:**
   If you are using SELinux or AppArmor, ensure that the policies are not preventing access to the file. You may need to adjust the policies or disable them temporarily to test.

5. **Logs and Debugging:**
   Check the logs for more detailed error messages that might give additional clues. You can increase the verbosity of the logs by adjusting the logging configuration in Authelia.

6. **File Creation:**
   Ensure that the file exists before starting Authelia. If Authelia is expected to create the file, ensure that the directory has the correct permissions.

If you have followed these steps and are still encountering issues, it might be helpful to provide more detailed logs or configuration snippets for further assistance.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
You will need to use `chmod` and `chown` to give authelia write access to the files. See the AI response which includes this information.

**Comment by mc962** [comment]
Good bot, following the instructions above _mostly_ worked. As an extra note, following the instructions about specifying the 
 user in the `systemd` file made it so it didn't have permissions to access the authelia configuration file, so I had to update permissions for that as well, following similar instructions.

It might be good to mention that in the documentation for this Notification Provider, as even though I knew that it was a permissions issue I didn't quite get the permissions command correct when I was trying on my own before.
